copy.go

```diff
@@ -54,6 +56,40 @@ type CopyOptions struct {
 	MapRoot func(ctx context.Context, src content.Storage, root ocispec.Descriptor) (ocispec.Descriptor, error)
 }
 
+// selectPlatform implements platform filter and returns the descriptor of
+// the first matched manifest from the manifest list / image index.
+func selectPlatform(ctx context.Context, src content.Storage, root ocispec.Descriptor, p *ocispec.Platform) (ocispec.Descriptor, error) {
+	if root.MediaType != docker.MediaTypeManifestList && root.MediaType != ocispec.MediaTypeImageIndex {
+		return ocispec.Descriptor{}, errdef.ErrUnsupported
```

Comment:
nit: We may want to elaborate the error message a bit, so that callers can know which node and which media type is not supported here.

Reply:
FYI: use [%w](https://golangbyexample.com/wrapping-and-unwrapping-error-golang/) to do the wrapping

Reply:
Made requested change